Back-To-School/Assets/BackToSchool/Scripts/GameManagement/GameManager.cs

```diff
@@ -10,155 +8,48 @@ namespace Assets.BackToSchool.Scripts.GameManagement
-            _player             = _playerObject.GetComponent<Player.Player>();
-            _player.PlayerStats = new PlayerStats();
-            _player.LevelSystem = new LevelSystem();
-
-            _statsManager = new StatsManager(_player);
-            _saveSystem   = new SaveSystem(_player, _statsManager);
-            _mainCamera.GetComponent<CameraFollow>().SetTarget(_playerObject.transform);
-            _inputManager = Instantiate(_inputManagerPrefab, transform.position, Quaternion.identity).GetComponent<InputManager>();
-            _inputManager.SetCamera(_mainCamera);
-            _enemySpawner = Instantiate(_enemySpawnerPrefab, transform.position, Quaternion.identity).GetComponent<EnemySpawner>();
-            _enemySpawner.SetTarget(_playerObject);
-        }
-
-        #region PlayerHandlers
-
-        private void OnPlayerReloaded()
-        {
-            if (!(_isPlayerDead || _isGamePaused)) _player.WeaponController.Reload();
-        }
-
-        private void OnNextWeapon(bool isNext) => _player.WeaponController.NextWeapon(isNext);
-
-        private void OnPlayerFire()
-        {
-            if (!(_isPlayerDead || _isGamePaused)) _player.Fire();
-        }
+        public bool IsSaveDataExists() => _saveSystem != null && _saveSystem.IsSaveDataExists();
 
-        private void OnPlayerStop() => _player.Stop();
+        public void ExitGame() { Application.Quit(); }
 
-        private void OnPlayerRotate(Vector3 pointToRotate)
-        {
-            if (!(_isPlayerDead || _isGamePaused)) _player.Rotate(pointToRotate);
-        }
+        public void StartGame(GameParameters parameters) => StartCoroutine(LoadGame(parameters));
+        public void ReturnToMenu()                       => SceneManager.LoadScene(Constants.SceneNames.MainMenu.ToString());
+        public void RestartLevel(string sceneName)       => StartGame(new GameParameters(true, sceneName));
 
-        private void OnPlayerMove(Vector3 direction)
+        public IEnumerator LoadGame(GameParameters parameters)
         {
-            if (!(_isPlayerDead || _isGamePaused)) _player.Move(direction);
-        }
+            AddParametersForLevel(parameters);
+            var asyncOp = SceneManager.LoadSceneAsync(parameters.NextScene);
+            while (!asyncOp.isDone) { yield return null; }
 
-        private void OnPlayerDeath()
-        {
-            _isPlayerDead = true;
-            _enemySpawner.SetTarget(null);
-            Invoke(nameof(EndGame), _gameOverDelay);
+            var game = FindObjectOfType<Game>();
+            game.Initialize(_saveSystem, Instance, parameters);
         }
 
-        #endregion
-
-        #region GameHandlers
-
-        private void StopGame()
+        private void Awake()
         {
-            Time.timeScale = 0f;
-            _isGamePaused  = true;
-            _pausePresenter.TogglePausePanel(_isGamePaused);
-        }
+            _saveSystem = new SaveSystem();
```

Comment:
Place under singletone implementation. Also, if destroying - do not call any other code